crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -187,14 +187,23 @@ def nodes
         # At this point params[:component] should be a node, if it is not,
         # raise an error.
         nodes_names = params[:component].split(/[\s,;]/)
+        upgraded_node_names = []
         nodes_names.each do |name_or_alias|
           node = ::Node.find_node_by_name_or_alias(name_or_alias)
-          next unless node.nil?
+          if node.nil?
+            raise ::Crowbar::Error::UpgradeError,
+              "Component must be 'all', 'controllers', 'resume', "\
+              "'postpone' or a node name(s). "\
+              "No node with '#{name_or_alias}' name or alias was found. "
+          end
+          upgraded_node_names << name_or_alias if node.upgraded?
+        end
+
+        if (node_names - upgraded_node_names).empty?
```

Comment:
It's `nodes_names` at line 189

Reply:
renamed all `*node*` to `*nodes*` for consistency.